File: app/models/order.py
from sqlalchemy import DECIMAL, TIMESTAMP, text, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy.dialects.postgresql import UUID
from models import Base
import uuid
from datetime import datetime
from decimal import Decimal

class Order(Base):
    __tablename__ = "orders"

    order_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()"))
    customer_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("customer.customer_id"), nullable=False) 
    user_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("users.user_id"), nullable=False)
    order_total: Mapped[Decimal] = mapped_column(DECIMAL(10, 2), nullable=False, default=0.00)
    status_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("status.status_id"), nullable=False)
    order_date: Mapped[datetime] = mapped_column(TIMESTAMP, nullable=False, server_default=text("CURRENT_TIMESTAMP"))
    updated_at: Mapped[datetime] = mapped_column(TIMESTAMP, server_default=text("CURRENT_TIMESTAMP"))

    customer = relationship("Customer")
    user = relationship("User")
    status = relationship("Status")

File: frontend/src/pages/orders/types/order.ts
type Status = {
  statusId: string;
  statusCode: string;
};

export type Customer = {
  customerId: string;
  customerName: string;
  customerPhone: string;
  customerEmail: string;
};

type User = {
  userId: string;
  userName: string;
};

export type Order = {
  orderId: string;
  orderTotal: string;
  orderDate: string;
  orderAttachment: string | null;
  updatedAt: string;
  status: Status;
  customer: Customer;
  user: User;
};

export type OrdersResponse = {
  orders: Order[];
  prevCursorDate: string | null;
  prevCursorId: string | null;
  nextCursorDate: string | null;
  nextCursorId: string | null;
  totalPages: number;
  currentPage: number;
  totalOrders: number;
  ordersPerPage: number;
};